### PR TITLE
RIA-8243 Update Tribunal Decision R31 Notification

### PR DESCRIPTION
**Pull request by javierhmcts**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/RIA-8243

### Change description ###

Update Tribunal Decision R31 Notification

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -927,6 +927,15 @@ govnotify:
           sms: 24dd4193-554b-40be-a0b3-56b0f80e37ab
         respondent:
           email: 6ebfdf22-9d9c-468e-a7cb-0fa84a7986e9
+    updateTribunalDecision:
+      rule31:
+        legalrep:
+          beforeListing:
+            email: da39849d-58cd-4e91-ab79-ea3083acd753
```
Comment (ThomasKKC) [inline src/main/resources/application.yaml:934]: We don't need to show Listing reference anymore, so we don't need templates here

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java [hunk 2]
```diff
@@ -3858,6 +3826,26 @@ && isDlrmSetAsideEnabled(asylumCase)
...
+            @Qualifier("updateTribunalDecisionRule31NotificationGenerator")
+            List<NotificationGenerator> notificationGenerators) {
+
+        return new NotificationHandler(
+                (callbackStage, callback) -> {
+                    AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+                    return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+                            && callback.getEvent() == Event.UPDATE_TRIBUNAL_DECISION
+                            && isDlrmSetAsideEnabled(asylumCase)
```
Comment (ThomasKKC) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java:3840]: Don't need this check, as the event is new.
Can you also add !AIP journey? Make sure we don't send this during AIP journey.

File: src/functionalTest/resources/scenarios/RIA-8243-update-tribunal-decision-r31-notification.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  "request": {
+    "uri": "/asylum/ccdAboutToSubmit",
+    "credentials": "Judge",
+    "input": {
+      "id": 8243,
+      "eventId": "updateTribunalDecision",
+      "state": "ftpaDecided",
+      "caseData": {
+        "template": "minimal-appeal-submitted.json",
+        "replacements": {
```
Comment (ThomasKKC) [inline src/functionalTest/resources/scenarios/RIA-8243-update-tribunal-decision-r31-notification.json:12]: You sure this work? there is no "updateTribunalDecisionList": "underRule31" ?

File: src/test/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeUpdateTribunalDecisionRule31PersonalisationEmailTest.java [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+        assertThatThrownBy(() -> legalRepresentativeUpdateTribunalDecisionRule31PersonalisationEmail.getPersonalisation((Callback<AsylumCase>) null))
+            .isExactlyInstanceOf(NullPointerException.class)
+            .hasMessage("callback must not be null");
+    }
+
+    @Test
+    void should_return_personalisation_first_check_when_all_information_given() {
+
+        DynamicList dynamicList = new DynamicList(new Value("allowed", "Yes, change decision to Dismissed"),
```
Comment (ThomasKKC) [inline src/test/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeUpdateTribunalDecisionRule31PersonalisationEmailTest.java:107]: I think these values are incorrect, please check UpdateTribunalDecisionRule31MidEvent in ia-case-api

**Review by ThomasKKC — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java [hunk 5]
```diff
@@ -3858,6 +3826,26 @@ && isDlrmSetAsideEnabled(asylumCase)
...
+    public PreSubmitCallbackHandler<AsylumCase> updateTribunalDecisionRule31NotificationHandler(
+            @Qualifier("updateTribunalDecisionRule31NotificationGenerator")
+            List<NotificationGenerator> notificationGenerators) {
+
+        return new NotificationHandler(
+                (callbackStage, callback) -> {
+                    AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+                    return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+                            && callback.getEvent() == Event.UPDATE_TRIBUNAL_DECISION
```
Comment (MiguelMartinRojas) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java:3839]: Is this a brand new event?

**Review by ThomasKKC — APPROVED** [review]

**Review by MiguelMartinRojas — APPROVED** [review]

**Pull request merged**